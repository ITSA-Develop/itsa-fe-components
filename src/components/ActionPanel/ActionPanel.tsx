import { mdiDotsVertical } from '@mdi/js';
import Icon from '@mdi/react';
import { Button, Dropdown, MenuProps } from 'antd';

export interface IActionPanel {
	items: MenuProps['items'];
}

export const ActionPanel = ({ items }: IActionPanel) => {
	return (
		<Dropdown menu={{ items }} trigger={['click']} placement="bottomRight">
			<Button icon={<Icon path={mdiDotsVertical} size={1} />} type="text" shape="circle" />
		</Dropdown>
	);
};
