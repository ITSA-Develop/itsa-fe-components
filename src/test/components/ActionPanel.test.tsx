import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { describe, expect, it } from 'vitest';

import { ActionPanel } from '../../components/ActionPanel/ActionPanel';

describe('ActionPanel', () => {
	const items = [
		{ key: '1', label: 'Editar' },
		{ key: '2', label: 'Eliminar' },
	];

	it('renders button with icon', () => {
		const { container } = render(<ActionPanel items={items} />);
		const button = screen.getByRole('button');
		expect(button).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	it('shows menu items when button is clicked', async () => {
		const { container } = render(<ActionPanel items={items} />);
		const user = userEvent.setup();

		items.forEach(item => {
			expect(screen.queryByText(item.label)).not.toBeInTheDocument();
		});

		const button = screen.getByRole('button');
		await user.click(button);

		items.forEach(item => {
			expect(screen.getByText(item.label)).toBeInTheDocument();
		});

		expect(container).toMatchSnapshot();
	});
});
