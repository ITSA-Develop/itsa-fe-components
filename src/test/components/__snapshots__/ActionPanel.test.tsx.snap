// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActionPanel > renders button with icon 1`] = `
<div>
  <button
    class="ant-btn css-dev-only-do-not-override-l9pxc0 ant-btn-circle ant-btn-text ant-btn-color-default ant-btn-variant-text ant-btn-icon-only ant-dropdown-trigger"
    type="button"
  >
    <span
      class="ant-btn-icon"
    >
      <svg
        role="presentation"
        style="width: 1.5rem; height: 1.5rem;"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,16A2,2 0 0,1 14,18A2,2 0 0,1 12,20A2,2 0 0,1 10,18A2,2 0 0,1 12,16M12,10A2,2 0 0,1 14,12A2,2 0 0,1 12,14A2,2 0 0,1 10,12A2,2 0 0,1 12,10M12,4A2,2 0 0,1 14,6A2,2 0 0,1 12,8A2,2 0 0,1 10,6A2,2 0 0,1 12,4Z"
          style="fill: currentColor;"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ActionPanel > shows menu items when button is clicked 1`] = `
<div>
  <button
    class="ant-btn css-dev-only-do-not-override-l9pxc0 ant-btn-circle ant-btn-text ant-btn-color-default ant-btn-variant-text ant-btn-icon-only ant-dropdown-trigger ant-dropdown-open"
    type="button"
  >
    <span
      class="ant-btn-icon"
    >
      <svg
        role="presentation"
        style="width: 1.5rem; height: 1.5rem;"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,16A2,2 0 0,1 14,18A2,2 0 0,1 12,20A2,2 0 0,1 10,18A2,2 0 0,1 12,16M12,10A2,2 0 0,1 14,12A2,2 0 0,1 12,14A2,2 0 0,1 10,12A2,2 0 0,1 12,10M12,4A2,2 0 0,1 14,6A2,2 0 0,1 12,8A2,2 0 0,1 10,6A2,2 0 0,1 12,4Z"
          style="fill: currentColor;"
        />
      </svg>
    </span>
  </button>
</div>
`;
